pydantic/dataclasses.py

```diff
@@ -197,8 +201,10 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         cls.__doc__ = original_doc
         cls.__module__ = original_cls.__module__
         cls.__qualname__ = original_cls.__qualname__
-        _pydantic_dataclasses.set_dataclass_fields(cls)
```

Comment:
I moved this into the `complete_dataclass` method, so that the fields would also be updated when doing a rebuild.